### ISSUE TITLE
Error in mimetype file - but it contains right string

### ISSUE THREAD

**Issue by robogeek** [IDPF/epubcheck#530]
I've written a tool, AkashaEPUB, to generate EPUB3's and am using epubcheck as one of the validation tools that I've done the right thing.

The only message being printed is:

```
$ java -jar /usr/bin/epubcheck guide.epub 
Epubcheck Version 3.0.1

ERROR: guide.epub: Mimetype contains wrong type (application/epub+zip expected).
Validating against EPUB version 3.0

Check finished with warnings or errors
```

Funny thing is the mimetype file contains the expected text.

```
$ cat out/mimetype 
application/epub+zip
```

The file has ONLY that text - no line feeds etc.  If I add a line feed, there's another message printed about how the file must ONLY have this mimetype text and nothing else.

The file is written with Node.js code like this:

```
fs.writeFile(path.join(config.root_out, "mimetype"), "application/epub+zip", "utf8", next);
```

In other words, it's telling the OS to make it a UTF-8 file, in case that makes any difference.

Finally, for more detail:

```
$ od -c out/mimetype 
0000000   a   p   p   l   i   c   a   t   i   o   n   /   e   p   u   b
0000020   +   z   i   p
0000024
```

Scratching my head and not seeing the problem.

I've been ignoring the issue - but the Apple iTunes book store is rejecting my EPUB because of this Error.

**Comment by mattgarrish** [comment]
Sounds like you’re probably getting extra headers when you zip the epub. Does the mimetype file name start at byte 30 and the value at byte 38?

 

Matt

**Comment by robogeek** [comment]
Ah...  Here's a dump of the start of the the EPUB contents.  The filename is at byte 30, and that 'K' is at byte 38.  I suppose it's compressed hence the unrecognizable characters.

```
0000000   P   K 003 004 024  \0  \b  \0  \b  \0 366  \b 235   F  \0  \0
0000020  \0  \0  \0  \0  \0  \0  \0  \0  \0  \0  \b  \0  \0  \0   m   i
0000040   m   e   t   y   p   e   K   ,   ( 310 311   L   N   , 311 314
0000060 317 323   O   -   (   M 322 256 312   ,  \0  \0   P   K  \a  \b
0000100   o   a 253   , 026  \0  \0  \0 024  \0  \0  \0   P   K 003 004
0000120 024  \0  \b  \0  \b  \0 366  \b 235   F  \0  \0  \0  \0  \0  \0
0000140  \0  \0  \0  \0  \0  \0 026  \0  \0  \0   M   E   T   A   -   I
0000160   N   F   /   c   o   n   t   a   i   n   e   r   .   x   m   l
```

**Comment by robogeek** [comment]
Have checked some other EPUB's I have on hand, and the contents of the mimetype file are in the clear.  So I need to see how to add that file w/o compression, it seems.

**Comment by robogeek** [comment]
Yup, that was it.  Successfully changed my code to not compress that one file and now there are no errors.

SUGGESTION: Mebbe epubcheck could detect this condition and give a more meaningful error?

**Comment by vanarajan2008** [comment]
hi robogeek,

How was you change your code without compress?  

Can you please drop the code for me?

**Comment by robogeek** [comment]
It was fine, and I thought this issue had already been closed.
